File: packages/sdk-python/trustchain/documents.py
from __future__ import annotations

from typing import Any

from .client import TrustChainClient


class DocumentsResource:
    def __init__(self, client: TrustChainClient) -> None:
        self._client = client

    def create(self, payload: dict[str, Any], *, idempotency_key: str | None = None) -> dict[str, Any]:
        return self._client.request("POST", "/documents", body=payload, idempotency_key=idempotency_key)

    def get(self, document_id: str) -> dict[str, Any]:
        return self._client.request("GET", f"/documents/{document_id}")

File: packages/sdk-typescript/src/documents.ts
import type { TrustChainClient } from "./client.js";
import type { CreateDocumentInput, Document } from "./types.js";

export class DocumentsResource {
  constructor(private readonly client: TrustChainClient) {}

  create(input: CreateDocumentInput, opts?: { idempotencyKey?: string }) {
    return this.client.request<{ document: Document }>({
      method: "POST",
      path: "/documents",
      body: input,
      idempotencyKey: opts?.idempotencyKey,
    });
  }

  get(id: string) {
    return this.client.request<{ document: Document }>({
      method: "GET",
      path: `/documents/${id}`,
    });
  }
}
